catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -78,10 +83,19 @@
 
     private final AttributeRegistry attributeRegistry;
 
-    private static final int DEFAULT_PAGE_SIZE = 10;
-
     private static final String ISO_8601_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
 
+    static {
+        Integer solrQueryRowSize;
+        try {
+            solrQueryRowSize = Integer.valueOf(System.getProperty("solr.query.rows", Integer.toString(2_000_000_000)));
+        } catch (NumberFormatException e) {
+            solrQueryRowSize = 2_000_000_000;
```

Comment:
Right now this is only being used in one place.  re : `This query is called to get the associated Metacards by a Metacard ID. I'm not sure the amount of associated Metacards would ever get to this point. To avoid blowing up SolrCloud, it may be more reasonable to reduce this page size to a few thousand records.`